lib/statsig/user.ex

```diff
@@ -0,0 +1,60 @@
+defmodule Statsig.User do
+  @derive {Jason.Encoder, only: [
+    :userID,
+    :customIDs,
+    :email,
+    :ip,
+    :userAgent,
+    :country,
+    :locale,
+    :appVersion,
+    :custom,
+    :statsigEnvironment
+  ]}
+
+  @type custom_value :: String.t() | number() | boolean() | nil
+  @type custom_attributes :: %{optional(String.t()) => custom_value()}
+
+  @type t :: %__MODULE__{
+    userID: String.t() | nil,
+    customIDs: %{optional(String.t()) => String.t()} | nil,
+    email: String.t() | nil,
+    ip: String.t() | nil,
+    userAgent: String.t() | nil,
+    country: String.t() | nil,
+    locale: String.t() | nil,
+    appVersion: String.t() | nil,
+    custom: custom_attributes() | nil,
+    privateAttributes: custom_attributes() | nil,
```

Comment:
it'd be better to default custom and private attributes to empty maps and ensure that they're always set to be maps. It'll save you a bunch on nil checks later.